packages/@react-aria/virtualizer/src/Virtualizer.tsx

```diff
@@ -135,13 +140,14 @@ export function useVirtualizer<T extends object, V, W>(props: VirtualizerOptions
     }
 
     lastFocusedKey.current = focusedKey;
-  }, [focusedKey, virtualizer.visibleRect.height, virtualizer, lastFocusedKey, scrollToItem]);
+  }, [focusedKey, virtualizer.visibleRect.height, virtualizer, lastFocusedKey, scrollToItem, modality]);
```

Comment:
Do we want this to re-run when the modality changes, even if the focused key didn't change? It will also cause all virtualizers on the page to re-render. Did you try using `getInteractionModality()` above instead of the hook?

Reply:
Yeah, I had tried that before apparently: https://github.com/adobe/react-spectrum/commit/518463f126ae0430864345a1320d910c2bd92c84 but I can't make heads or tails of what exactly the "shift tab behavior" that commit refers to, everything seems ok to me... I'll go back to using `getInteractionModality`